packages/gatsby/src/schema/resolvers.ts

```diff
@@ -29,6 +28,9 @@ import { IGatsbyNode } from "../redux/types"
 
 type ResolvedLink = IGatsbyNode | Array<IGatsbyNode> | null
 
+type nestedListOfStrings = Array<string | nestedListOfStrings>
+type nestedListOfNodes = Array<IGatsbyNode | nestedListOfNodes>
```

Comment:
Funsies. But turns out the resulting nesting structure must match the input nesting structure. So if you have `[a, [b, [c, d]]]` as input, then your output should be the same except the input strings are replaced with nodes.

This is a tad annoying since it means we can't "just" pass on a flat array of promises to `Promise.all()` but so be it.

Reply:
I realise this is already merged, but types should be PascalCased